Add Linux Commands deprecation document

mbbx6spp opened the pull request.

mbbx6spp commented:
See this for formatted/presented ASCIIDOC output: https://github.com/mbbx6spp/styleguides/blob/shell-command-deprecations/bash/COMMANDS.asciidoc

mbbx6spp commented:
@polynomial what other use case examples would be useful for `ss` command?

polynomial commented:
The only other one that is pretty common for me is -p, to show whats binding a port (netstat -anp | grep LISTEN replacement)

It's unclear why some of the examples have -r and -n, aren't these contradictory?

Include `nslookup` in deprecated list, provide examples for:
* host HOSTNAME
* dig -t ANY hostname

wan't to be fancy pants? Include dnstracer example for looking up the tree of authority without multiple commands.

polynomial commented on bash/COMMANDS.asciidoc:
```diff
@@ -0,0 +1,154 @@
+If you are still using `ifconfig` in your scripts for whatever reason today
+you should consult this GIF and meditate on it for some time:
+
+image::http://giphy.com/gifs/wWEDxkj0qzjLa["Ashamed"]
```
This doesn't work as an image because they do some bullshit detection stuff and try to feed you a full page i think...

mbbx6spp replied:
Ahh, thanks. I thought it was just GitHub's weird ASCIIDOC rendering.

mbbx6spp replied:
I just put a link now rather than attempting to embed an image.

mbbx6spp commented:
I will add the `-p` case for netstat and explain the `-r` and `-n` parts better. It's mostly a personal preference but I usually don't want to see `:documentum` instead of `:10002` or on overloaded ports `*:wap-wsp` instead of `*:9200` for Elasticsearch, but I usually want to see the resolved hostnames instead of IPs. I will illustrate with examples.

mbbx6spp commented:
Note: I just added sections on `ip neighbor` and `ip link` this morning as replacements for `arp` and other `ifconfig` usages.

mbbx6spp commented:
@polynomial I think I addressed the `-p` use case and the explanation of the `-n` option and why you would use it over not using it in the new subsection titled 'Look ma, no pipes' https://github.com/mbbx6spp/styleguides/blob/shell-command-deprecations/bash/COMMANDS.asciidoc#-look-ma-no-pipes-

mbbx6spp commented:
Adding `nslookup` and alternatives now. Thanks!

mbbx6spp commented:
I added the `dig` section here: https://github.com/mbbx6spp/styleguides/blob/shell-command-deprecations/bash/COMMANDS.asciidoc#dig

Do we really need `dnstracer` which is non-standard command? (See last dig command example) :)

polynomial commented:
OMG THIS IS ALL GREAT AND I LOVE IT SO MUCH.

mbbx6spp commented:
WOW. SO CAPS, MUCH OMG.

mbbx6spp commented:
As per @stephenweber I updated the structure of the COMMANDS document to show the outline more as it was confusing and added the 'Alternatives' level.
